### PR TITLE
feat(typescript): upgrade to typescript 2.7.2

### PR DESCRIPTION
**Pull request by daviwil**
/cc @damieng

### CODE REVIEW

File: src/test/typescript-service-helpers.ts [hunk 1]
```diff
@@ -382,7 +382,7 @@ export function describeTypeScriptService(
                             character: 8,
                         },
                     },
-                    contents: [{ language: 'typescript', value: 'import Foo' }, '**alias**'],
+                    contents: [{ language: 'typescript', value: 'class Foo\nimport Foo' }, '**alias**'],
```
Comment (daviwil) [inline src/test/typescript-service-helpers.ts]: Looks like hovers now give you more symbol info than just the import statement!

**Comment by daviwil** [comment]
Anything I can/should do about the drop in code coverage?

**Comment by felixfbecker** [comment]
Hmm, seems like `getNewLine()` is not called anymore by TS. Could you check if it is still part of the interface? If not, we can safely remove it

https://codecov.io/gh/sourcegraph/javascript-typescript-langserver/pull/431/changes#L95

**Comment by daviwil** [comment]
Still part of the interface: https://github.com/Microsoft/TypeScript/blob/v2.7.2/src/services/types.ts#L163

Looks like there was a recent change to prioritize the use of `FormatCodeSettings` if it's provided before ever calling `getNewLine()`, so that could account for the lack of calls:

https://github.com/Microsoft/TypeScript/pull/21158/files#diff-183e2291cbb94d65ae0406354ce1ba7aR1262

I'll remove that function from the language service host and see what happens!

**Comment by felixfbecker** [comment]
In that case just leave it in and not worry about the coverage

**Comment by daviwil** [comment]
Ok, sounds good!

**Pull request merged**